plone/schemaeditor/browser/schema/delete_fieldset.py

```diff
@@ -0,0 +1,43 @@
+from plone.schemaeditor import _
+from plone.schemaeditor.utils import SchemaModifiedEvent
+from plone.supermodel.interfaces import FIELDSETS_KEY
+from Products.Five import BrowserView
+from Products.statusmessages.interfaces import IStatusMessage
+from zope.container.contained import notifyContainerModified
+from zope.event import notify
+
+
+class DeleteFieldset(BrowserView):
+
+    def __call__(self):
+        fieldset_name = self.request.form.get('name')
+        schema = self.context.schema
+        fieldsets = schema.queryTaggedValue(FIELDSETS_KEY, [])
+
+        new_fieldsets = []
+        for fieldset in fieldsets:
+            if fieldset.__name__ == fieldset_name:
+                if fieldset.fields:
+                    IStatusMessage(self.request).addStatusMessage(
+                        _(u'Only empty fieldsets can be deleted'),
+                        type='error')
+                    return self.request.RESPONSE.redirect(self.nextURL)
+                continue
+            else:
```

Comment:
The else is s bit confusing and superfluous, b/c above it is returned or continued.

Reply:
You mean the "continue" is superfluos, right? Without the "else" branch, all the fieldsets would be deleted.

Well, the problem is that we have to cycle through the fieldsets to check whether the one to delete is empty. Maybe we could do that check separately and then afterwards construct the new list of fieldsets with a list comprehension... That may be more readable. Somewhat like this:

```
        for fieldset in fieldsets:
            if fieldset.__name__ == fieldset_name and fieldset.fields:
                    IStatusMessage(self.request).addStatusMessage(
                        _(u'Only empty fieldsets can be deleted'),
                        type='error')
                    return self.request.RESPONSE.redirect(self.nextURL)
        new_fieldsets = [fiedset for fieldset in fieldsets if fieldset.__name__ != fieldset_name]
```

Reply:
No, I meant the else at all (and dedent the line after the else). Anyway, keep it as it is, its just cosmetics.

Reply:
Ah! Now I get it. Only the "else", not the whole branch. Well, I like my new version anyway. Might do another small PR. :-)

Thanks for reviewing, btw! <3